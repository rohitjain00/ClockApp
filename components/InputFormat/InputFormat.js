import React from 'react';
import {Linking, StyleSheet, Text, TextInput, View} from 'react-native';
import PropTypes from 'prop-types';

export function InputFormat(props) {
    function changeText(text) {
        props.onChange(text);
    }

    return (
        <View style={styles.container}>
            <Text
                onPress={() => Linking.openURL('https://momentjs.com/docs/#/displaying/')}>
                {props.text}
            </Text>
            <TextInput
                style={styles.input}
                onChangeText={(text) => changeText(text)}
                defaultValue={props.value}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        marginTop: 20,
    },
});

InputFormat.propTypes = {
    onChange: PropTypes.func,
    text: PropTypes.string,
    value: PropTypes.string
}