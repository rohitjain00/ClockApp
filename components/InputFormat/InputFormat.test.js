//CustomColorPicker.test.js
import React from 'react';
import {render} from '@testing-library/react-native';
import { InputFormat } from "./InputFormat";
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('InputFormat Component', () => {
    const mockOnPress = jest.fn();
    const component = <InputFormat onChange={mockOnPress} text={'testing component'} value={'HH:MM"YYYY'}/>
    let wrapper;
    beforeEach(() => {
        wrapper = mount(component);
        jest.clearAllMocks();
    })
    it('Input Format Props Validation', async () => {
        const componentProps = wrapper.props();
        expect(componentProps.onChange).toEqual(mockOnPress);
        expect(componentProps.text).toEqual('testing component');
        expect(componentProps.value).toEqual('HH:MM"YYYY');
    });
    it('Input Format Snapshot Validation', async () => {
        const tree = render(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Input Format Action Validation', async () => {
        // https://github.com/enzymejs/enzyme/issues/2263#issuecomment-568585047
        // According to above statement We cannot interact with state inside a function with useState.
    });
    it('Input Format onChange handler', async () => {
        wrapper.props().onChange('hh:mm:yyyy');
        expect(mockOnPress).toHaveBeenCalled();
        expect(mockOnPress).toHaveBeenCalledTimes(1);
        wrapper.props().onChange('hh:ss');
        expect(mockOnPress).toHaveBeenCalledTimes(2);
    });
});