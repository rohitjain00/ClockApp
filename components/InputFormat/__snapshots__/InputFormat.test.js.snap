// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFormat Component Input Format Snapshot Validation 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
>
  <Text
    onPress={[Function]}
  >
    testing component
  </Text>
  <TextInput
    allowFontScaling={true}
    defaultValue="HH:MM\\"YYYY"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
  />
</View>
`;
